chains/evm/contracts/CCVRampProxy.sol

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import {ITypeAndVersion} from "@chainlink/contracts/src/v0.8/shared/interfaces/ITypeAndVersion.sol";
+
+/// @notice CCVRampProxy enables upgrades to CCVOnRamps and CCVOffRamps without breaking existing references in token pools, receivers, and apps.
+/// @dev All future versions of ICCVOnRamp and ICCVOffRamp must maintain remoteChainSelector, version, and caller as the first three parameters in every method.
+abstract contract CCVRampProxy is ITypeAndVersion {
+  error InvalidRemoteChainSelector(uint64 remoteChainSelector);
+  error InvalidVersion(bytes32 version);
+  error RampNotFound(uint64 remoteChainSelector, bytes32 version);
+
+  event RampSet(uint64 indexed remoteChainSelector, bytes32 indexed version, address indexed rampAddress);
+
+  struct SetRampsArgs {
+    uint64 remoteChainSelector; // ─╮ The remote chain selector.
+    address rampAddress; // ────────╯ The address of the ramp contract.
+    bytes32 version; // The version of the ramp.
+  }
+
+  string public constant override typeAndVersion = "CCVRampProxy 1.7.0-dev";
+
+  /// @notice The supported ramps.
+  /// @dev Each remote chain selector can have multiple ramps, each with a different version. This protects in-flight messages during upgrades.
+  mapping(uint64 remoteChainSelector => mapping(bytes32 version => address rampAddress)) private s_ramps;
```

Comment:
I think we can drop the entire `version` thing and make this a lot more efficient/smaller